### PR TITLE
RN-185 Fix left channel and teams while disconnected

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
When getting all the channels and members from the server we're going to remove the `myMembers` keys in the **channel** store that we already have just to make sure that those that were deleted aren't kept in the store

#### Ticket Link
https://mattermost.atlassian.net/browse/RN-185

### CODE REVIEW

File: src/actions/channels.js [hunk 1]
```diff
@@ -11,6 +11,17 @@ import {Client4} from 'client';
...
+function getChannelsIdForTeam(state, teamId) {
+    const {channels} = state.entities.channels;
+
+    return Object.values(channels).reduce((res, channel) => {
+        if (channel.team_id === teamId) {
+            res.push(channel.id);
+        }
+        return res;
+    }, []);
+}
```
Comment (jwilander) [inline src/actions/channels.js]: Can we throw this in `channel_utils.js`?

Comment (enahum) [inline src/actions/channels.js]: Will do

**Comment by enahum** [comment]
@jwilander done

**Comment by jwilander** [comment]
Looks good but you have a build failure

**Comment by enahum** [comment]
yeah, i'm rebuilding, the failure was cause of a timeout in something completely unrelated to this PR

**Pull request merged**